src/lib/hover_thrust_estimator/ZeroOrderHoverThrustEkf.cpp

```diff
@@ -0,0 +1,94 @@
+#include "ZeroOrderHoverThrustEkf.hpp"
+#include <mathlib/mathlib.h>
+
+void ZeroOrderHoverThrustEkf::predict(const float dt)
+{
+	// State is constant
+	// Predict state covariance only
+	_P += _Q * dt;
+	_dt = dt;
+}
+
+void ZeroOrderHoverThrustEkf::fuseAccZ(const float acc_z, const float thrust)
+{
+	const float H = computeH(thrust);
+	const float innov_var = computeInnovVar(H);
+	const float innov = computeInnov(acc_z, thrust);
+	const float K = computeKalmanGain(H, innov_var);
+	const float innov_test_ratio = computeInnovTestRatio(innov, innov_var);
+
+	printf("AZ = %.3f\tthr = %.3f\tH = %.3f\tinnov_var = %.3f\tinnov = %.3f\tK = %.3f\ttest_ratio = %.3f\n",
+	       (double)acc_z,
+	       (double)thrust,
+	       (double)H,
+	       (double)innov_var,
+	       (double)innov,
+	       (double)K,
+	       (double)innov_test_ratio);
+
+	float residual = innov;
+
+	if (isTestRatioPassing(innov_test_ratio)) {
+		updateState(K, innov);
+		updateStateCovariance(K, H);
+		residual = computeInnov(acc_z, thrust); // residual != innovation since the hover thrust changed
+		printf("estimated hover thrust: %.3f\t std_dev = %.3fres = %.3f\n",
+		       (double)_hover_thr,
+		       (double)sqrtf(_P),
+		       (double)residual);
+	}
+
+	updateMeasurementNoise(residual, H);
+}
+
+inline float ZeroOrderHoverThrustEkf::computeH(const float thrust)
+{
+	return -9.81f * thrust / (_hover_thr * _hover_thr);
```

Comment:
@bresch I think we have definitions for this.

Reply:
Right, I found it in `<ecl/geo/geo.h>`

Reply:
@bresch I mean gravity

Reply:
@bresch Sorry, did not see your answer, all good